CMakeLists.txt

```diff
@@ -0,0 +1,206 @@
+  else()
+    message(FATAL_ERROR "PCAP library not found!")
+  endif()
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_IMMEDIATE_MODE "Enable PCAP immediate Mode (supported on libpcap>=1.5)"
+       ${HAVE_PCAP_IMMEDIATE_MODE})
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_IMMEDIATE_MODE)
+  if(NOT HAVE_PCAP_IMMEDIATE_MODE)
+    message(FATAL_ERROR "PCAP library doesn't have Immediate Mode support!")
+  endif()
+  add_definitions(-DHAS_PCAP_IMMEDIATE_MODE)
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_SET_DIRECTION
+       "Enable set direction for capturing incoming or outgoing packets (supported on libpcap>=0.9.1)" OFF)
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_SET_DIRECTION)
+  if(NOT HAVE_PCAP_DIRECTION)
+    message(FATAL_ERROR "PCAP library doesn't have Direction support!")
+  endif()
+  add_definitions(-DHAS_SET_DIRECTION_ENABLED)
+endif()
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+if(PCAPPP_TARGET_PLATFORM_WINDOWS AND PCAPPP_TARGET_COMPILER_MSVC)
+  # Other hacks for VStudio
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+else()
+  if(NOT CMAKE_USE_PTHREADS_INIT)
+    message(FATAL_ERROR "Pthreads not found!")
+  endif()
+endif()
+
+if(PCAPPP_USE_DPDK)
+  find_package(DPDK)
+  if(NOT dpdk_FOUND)
+    message(FATAL_ERROR "Specify DPDK Library with env DPDK_DIR=<DPDK_PATH>")
+  endif()
+  add_definitions(-DUSE_DPDK)
+endif()
+
+# Git Commit and Branch
+add_compile_definitions(GIT_COMMIT="${GIT_HASH}")
+add_compile_definitions(GIT_BRANCH="${GIT_BRANCH}")
+
+if(PCAPPP_USE_PF_RING)
+  find_package(PF_RING)
+  if(NOT PF_RING_FOUND)
+    message(FATAL_ERROR "Packet library -DPF_RING_ROOT=<PF_RING_PATH>")
+  endif()
+  add_definitions(-DUSE_PF_RING)
+endif()
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release")
+endif()
+
+if(PCAPPP_TARGET_COMPILER_CLANG OR PCAPPP_TARGET_COMPILER_GCC)
+  add_compile_options(-Wall -g)
+endif()
+
+add_subdirectory(3rdParty)
+add_subdirectory(Packet++)
+add_subdirectory(Pcap++)
+add_subdirectory(Common++)
+
+if(PCAPPP_BUILD_EXAMPLES)
+  set(PCAPPP_BINARY_EXAMPLES_DIR ${CMAKE_BINARY_DIR}/examples_bin)
+  add_subdirectory(Examples)
+endif()
+
+if(PCAPPP_BUILD_TESTS)
+  include(CTest)
+  add_subdirectory(Tests)
+endif()
+
+if(PCAPPP_INSTALL)
+  set(CPACK_DEBIAN_PACKAGE_MAINTAINER "seladb")
+  set(CPACK_PACKAGE_FILE_NAME PcapPlusPlus-v${SEMVER})
+
+  include(CPack)
+
+  # Generate PKG-Config
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/PcapPlusPlus.pc.in ${CMAKE_CURRENT_BINARY_DIR}/PcapPlusPlus.pc @ONLY)
+  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/PcapPlusPlus.pc" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+  # CMake helpers to compile Pcap++ with CMake
+  if(LIGHT_PCAPNG_ZSTD)
```

Comment:
I have a comment that is not related to this file but to here: https://github.com/seladb/PcapPlusPlus/blob/22269e8d40bebdbef225d63bf0a92a9bd94988d7/3rdParty/LightPcapNg/CMakeLists.txt#L9

I think that the `LIGHT_PCAPNG_ZSTD` option should default to `OFF` and not `${ZSTD_FOUND}` because most users don't need the Zstd features. If we default it to `${ZSTD_FOUND}` people might need to link with the zstd library even if they don't need any of it features

Reply:
done